receptor_affinity/mesh.py

```diff
@@ -437,16 +456,14 @@ def validate_ping_results(results, threshold=0.1):
                 valid = False
         return valid
 
-    @staticmethod
-    def compare_routes(route1, route2):
-        if route1 != route2:
-            print(f"****====MATCH FAIL")
-            print(route1)
-            print(route2)
-            return False
-        else:
-            print("****====MATCH")
-            return True
-
-    def validate_all_node_routes(self):
-        return all(node.validate_routes() for node in self.nodes.values() if node.active)
+    def validate_routes(self) -> None:
+        problem_nodes = []
+        for node in self.nodes.values():
+            if not node.active:
+                continue
+            try:
+                node.validate_routes()
+            except RouteMismatchError:
+                problem_nodes.append(node)
+        if problem_nodes:
+            raise RouteMismatchError(self, problem_nodes)
```

Comment:
Do we want to distinguish between a Node route issue and a Mesh route issue?

Reply:
If a mesh and one or more of its nodes disagree about routes, then which is correct and which is incorrect?